tests/TestCase/ORM/Association/HasOneTest.php

```diff
@@ -79,6 +79,21 @@ public function tearDown()
     }
 
     /**
+     * Tests that foreignKey() returns the correct configured value
+     *
+     * @return void
+     */
+    public function testForeignKey()
+    {
+        $assoc = new HasOne('Profiles', [
+            'sourceTable' => $this->user
+        ]);
+        $this->assertEquals('user_id', $assoc->foreignKey());
+        $this->assertEquals('another_key', $assoc->foreignKey('another_key'));
+        $this->assertEquals('another_key', $assoc->foreignKey());
```

Comment:
In some places we haven't. I've been adding them for the request/response classes. But you're right that most of the new ORM methods don't have explicit tests.